lib/beaker/host/unix.rb

```diff
@@ -138,6 +138,23 @@ def external_copy_base
       @external_copy_base
     end
 
+    # Tells you whether a host platform supports beaker's
+    #   {Beaker::HostPrebuiltSteps#set_env} method
+    #
+    # @return [String,nil] Reason message if set_env is not supported,
+    #   nil if it is supported
+    def supports_set_env
```

Comment:
So I don't know that this is necessarily worth a lot of discussion (I'm not going to die on this hill). I grok your observation about negative methods, though I think the current method name and return values are just as confusing as the double negative instance.

If the method name can't be in line with the return value (truthy values answer "yes" to the question the method name asks) like `skip_setting_env()`.

Then I walk through the same path that you outlined, but I see it as, we're answering two questions with the one method, "do we skip setting the environment up for the platform?" and "why are we skipping setting up the environment?" I would argue that those two questions are similar enough that they don't warrant two method calls, or its own object[1], and would return two answers to those questions.

Again, not my hill. So feel free to ignore me. What do other's think? @colinPL ? @waynr ? (since Rick was already pinged).

1. unless this, and more complicated issues, are happening in many places and could be cleaned up with a Result object. But I would still put that out of scope of this PR.